utils/preferences/src/extensions/unicode/keywords/numbering_system.rs

```diff
@@ -0,0 +1,19 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::extensions::unicode::errors::Error;
+use crate::struct_keyword;
+use icu_locid::extensions::unicode::Value;
+use tinystr::TinyAsciiStr;
+
+struct_keyword!(NumberingSystem, "nu", TinyAsciiStr<8>, |input: &Value| {
+    //XXX: Perf
+    let s = input.to_string();
+    let v = TinyAsciiStr::from_str(&s);
```

Comment:
Suggestion: use the doc-hidden fn `input.as_single_subtag()`; you can consider making it public if you like